### PR TITLE
LOOP-1380: Loop Notifications settings screens

### PR DESCRIPTION
**Pull request by rickpasetto**
Includes:
- new button in top-level Settings to open "Loop Notifications" Settings
- Loop Notification Settings screen (which doesn't have much on it...this is where, eventually, all the device managers' notification settings would surface)
- NotificationsCriticalAlertsPermissions settings

NOTE: I'm posting this for PR review, but note that it is also in design review right now.  If I should hold this PR for now, please let me know.   

[LOOP-1380](https://tidepool.atlassian.net/browse/LOOP-1380)

### CODE REVIEW

File: Loop/View Controllers/SettingsTableViewController.swift [hunk 1]
```diff
@@ -194,6 +198,7 @@ final class SettingsTableViewController: UITableViewController {
                 }
                 cell.textLabel?.text = cgmManager.localizedTitle
                 cell.detailTextLabel?.text = nil
+                cell.accessoryType = .disclosureIndicator
```
Comment (nhamming) [inline Loop/View Controllers/SettingsTableViewController.swift:202]: [nit] curious a disclosure indicator is added here, but not for the pump settings.

File: LoopUI/Views/LoopNotificationsView.swift [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            Text(backButtonText)
+        }
+    }
+            
+    private var notificationAndCriticalAlertPermissionSection: some View {
+        Section(header: SectionHeader(label: LocalizedString("Tidepool Loop Notifications", comment: "Section title for Tidepool Loop notifications"))) {
+            NavigationLink(destination: notificationAndCriticalAlertPermissionScreen) {
+                Text(LocalizedString("Notification & Critical Alert Permissions", comment: "Notification & Critical Alert Permissions button text"))
+            }
+        }
```
Comment (nhamming) [inline LoopUI/Views/LoopNotificationsView.swift:54]: [nit] Just FYI, but it was the use of `NavigationLink` and the ripple effect of updating view state based on observed objects that caused a number of issues for the CGM settings. If the view state update triggers on regular occurring events, e.g. incoming CGM measurements, it is possible the view and view model will update unexpectedly.

**Review by nhamming — APPROVED** [review]
One FYI warning (however, it may not apply to these views). Otherwise LGTM

File: Loop.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -4118,7 +4134,7 @@
 		4F7528901DFE1DC600C322D6 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				APPLICATION_EXTENSION_API_ONLY = YES;
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (darinkrauss) [inline Loop.xcodeproj/project.pbxproj:4137]: Was this an intentional change on the `Loop` project? If so, what was the reason?

File: Loop.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -4133,7 +4149,7 @@
 		4F7528911DFE1DC600C322D6 /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				APPLICATION_EXTENSION_API_ONLY = YES;
+				APPLICATION_EXTENSION_API_ONLY = NO;
```
Comment (darinkrauss) [inline Loop.xcodeproj/project.pbxproj:4152]: Was this an intentional change on the `Loop` project? If so, what was the reason?

File: Loop/Managers/Alerts/InAppModalAlertPresenter.swift [hunk 5]
```diff
@@ -94,7 +94,9 @@ extension InAppModalAlertPresenter {
                 return
             }
             self.playSound(for: alert)
-            let alertController = self.presentAlert(title: content.title, message: content.body, action: content.acknowledgeActionButtonLabel) { [weak self] in
+            let alertController = self.presentAlert(title: content.title, message: content.body,
```
Comment (darinkrauss) [inline Loop/Managers/Alerts/InAppModalAlertPresenter.swift]: [nit] I would say either all on one line or all on separate lines?

File: LoopUI/Views/NotificationsCriticalAlertPermissionsView.swift [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+extension NotificationsCriticalAlertPermissionsView {
+    private var manageNotificationsSection: some View {
+        Section {
+            Button( action: { self.viewModel.gotoSettings() } ) {
+                HStack {
+                    Text(LocalizedString("Manage Notifications in Settings", comment: "Manage Notifications in Settings button text"))
+                    if !viewModel.notificationsPermissionsGiven {
+                        Text("⚠️")
```
Comment (darinkrauss) [inline LoopUI/Views/NotificationsCriticalAlertPermissionsView.swift]: [nit] LocalizedString?

Comment (darinkrauss) [reply LoopUI/Views/NotificationsCriticalAlertPermissionsView.swift]: [and others down below...]

File: LoopUI/Views/NotificationsCriticalAlertPermissionsViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            self?.updateState()
+        }
+        updateState()
+    }
+    
+    private func updateState() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            self.notificationsPermissionsGiven = settings.alertSetting == .enabled
+            self.criticalAlertsPermissionsGiven = settings.criticalAlertSetting == .enabled
+            print("notificationsPermissionsGiven = \(self.notificationsPermissionsGiven)")
```
Comment (darinkrauss) [inline LoopUI/Views/NotificationsCriticalAlertPermissionsViewModel.swift]: [nit] Remove `print` statements? Or turn them into `log` statements if needed.

**Review by darinkrauss — COMMENTED** [review]
A question and some nits...

File: Loop/View Controllers/SettingsTableViewController.swift (hunk 1 above)
Comment (rickpasetto) [reply Loop/View Controllers/SettingsTableViewController.swift:202]: Oversight on my part.  Good catch.

File: LoopUI/Views/LoopNotificationsView.swift (hunk 2 above)
Comment (rickpasetto) [reply LoopUI/Views/LoopNotificationsView.swift:54]: Hm.  Okay, good to know.  This all will need rework anyway (post-510k) so I am not too concerned.

File: Loop.xcodeproj/project.pbxproj (hunk 3 above)
Comment (rickpasetto) [reply Loop.xcodeproj/project.pbxproj:4137]: Yes.  I needed to do this in order to be able to call `UIApplication.shared.open()` in `NotificationsCriticalAlertPermissionsViewModel.swift`

File: Loop.xcodeproj/project.pbxproj (hunk 4 above)
Comment (rickpasetto) [reply Loop.xcodeproj/project.pbxproj:4152]: See above

**Review by darinkrauss — APPROVED** [review]
LGTM! 🍔

**Pull request merged**